test/base-benchmark.cpp

```diff
@@ -0,0 +1,51 @@
+/* Icinga 2 | (c) 2024 Icinga GmbH | GPLv2+ */
+
+#include "base/benchmark.hpp"
+#include "base/utility.hpp"
+#include <BoostTestTargetConfig.h>
+#include <chrono>
+#include <cmath>
+
+using namespace icinga;
+
+static bool AssertSumSeconds(Benchmark& sum, double seconds)
+{
+	return std::abs(((double)sum - seconds) / seconds) < 0.1;
+}
+
+BOOST_AUTO_TEST_SUITE(base_benchmark)
+
+BOOST_AUTO_TEST_CASE(zero)
+{
+	BOOST_CHECK_EQUAL((double)Benchmark(), 0);
+}
+
+BOOST_AUTO_TEST_CASE(one)
+{
+	Benchmark sum;
+
+	auto start (Benchmark::Clock::now());
+	Utility::Sleep(0.25);
+	sum += Benchmark::Clock::now() - start;
+
+	BOOST_CHECK(AssertSumSeconds(sum, 0.25));
+}
+
+BOOST_AUTO_TEST_CASE(two)
+{
+	Benchmark sum;
+
+	{
+		auto start (Benchmark::Clock::now());
+		Utility::Sleep(0.25);
+		sum += Benchmark::Clock::now() - start;
+	}
+
+	auto start (Benchmark::Clock::now());
+	Utility::Sleep(0.5);
+	sum += Benchmark::Clock::now() - start;
+
+	BOOST_CHECK(AssertSumSeconds(sum, 0.75));
+}
```

Comment:
Does doing `Benchmark::Clock::now() - start` actually still test anything useful? Like doesn't that part now only test that subtracting `std::chrono::steady_clock::now()` gives the time delta between two calls, so this is more a test for the STL than for the `Benchmark` class.